tests/test_codex_feed_hooks.py

```
@@ -524,8 +524,42 @@ def test_install_adds_codex_permission_request_hook(cli_path: str, root: Path) -
+
+def test_install_migrates_legacy_codex_hooks_feature(cli_path: str, root: Path) -> None:
+    codex_home = root / "codex-home-legacy"
+    codex_home.mkdir()
+    (codex_home / "config.toml").write_text(
+        "[features]\napps = true\ncodex_hooks = false\nhooks = false\n",
+        encoding="utf-8",
+    )
+    env = os.environ.copy()
+    env["CODEX_HOME"] = str(codex_home)
+
+    result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)